### ISSUE TITLE
x64dbg can't load DLL with no entrypoint

### ISSUE THREAD

**Issue by lhmouse** [x64dbg/x64dbg#3467]
### Operating System

Windows 11 23H2 22631.4317

### x64dbg Version

Oct 18 2024

### Describe the issue

It is possible to create a DLL without an entrypoint. x64dbg fails to load such a DLL, but if I call `LoadLibraryW()` to load it _with exactly the same path_, my program doesn't fail.

I actually tried compiling https://github.com/x64dbg/x64dbg/blob/9785d1a62e104a2720c21736db8fc8c6f7311ce3/src/loaddll/loaddll.cpp with the path hard-coded to my DLL but it did not fail. Not sure what is special about the prebuilt ones in x64dbg.

### Steps to reproduce

1. Create a source file:
    ```c
    // a.c
    int foo(void) { return 1; }
    ```
2. In MSYS2 UCRT64 shell, compile it with gcc:
    > gcc a.c -shared -nostdlib -o a.dll
    ld.exe: warning: cannot find entry symbol DllMainCRTStartup; not setting start address

3. Load this file with x64dbg. It says 'Failed to load DLL (LastError: 193)' which is `ERROR_BAD_EXE_FORMAT`.

**Comment by eternablue** [comment]
Hello I saw the issue has been open for a while, have you or anyone else started working on it? I would really like to contribute to it

**Comment by lhmouse** [comment]
Nobody is working on it I think.

**Comment by mrexodia** [comment]
Yeah feel free to work on any issue you like @eternablue! Just take a look at the current pull requests and make sure to not duplicate those.

**Comment by eternablue** [comment]
Hello, i have not managed to reproduce your issue, could you provide me with your compiled binary / check if you still have this issue ?

**Comment by lhmouse** [comment]
@eternablue This issue is still reproducible with x64dbg Mar 3 2025, Windows 11 24H2 26120.3380, Insider Preview, Dev channel.

![Image](https://github.com/user-attachments/assets/b1acfb41-15a0-4620-997d-860cebcdba56)
[a.dll.gz](https://github.com/user-attachments/files/19264791/a.dll.gz)

**Comment by eternablue** [comment]
Hello i have not managed to reproduce the issue on windows 10 22H2 19045.5608 even with your binary and i have not managed to install windows 11 on a VM therefore i won't be able to solve this issue...

**Comment by lhmouse** [comment]
Take it easy. It's not an important issue.

**Comment by lhmouse** [comment]
Yes `AddressOfEntryPoint` can only be zero for DLLs.

Reference: https://learn.microsoft.com/en-us/windows/win32/debug/pe-format#optional-header-standard-fields-image-only

**Comment by mrexodia** [comment]
No it can be zero for executables and everything will run fine 🙂 Don’t believe the specs, they are lies unfortunately…
